src/sql/src/func.rs

```diff
@@ -3927,7 +3927,14 @@ pub static MZ_INTERNAL_BUILTINS: Lazy<BTreeMap<&'static str, Func>> = Lazy::new(
                     unnest($2) AS search_schema (name)
                     ON search_schema.name = s.name
                 JOIN
-                    mz_catalog.mz_databases AS d
+                    (
+                        SELECT id, name FROM mz_catalog.mz_databases
+                        -- If the provided database does not exist, add a row for it so that it
+                        -- can still join against ambient schemas.
+                        UNION ALL
+                        SELECT '', $1
+                        WHERE (SELECT $1 NOT IN (SELECT name FROM mz_catalog.mz_databases))
```

Comment:
This can probably actually be simplified to `WHERE $1 NOT IN (SELECT name FROM mz_catalog.mz_databases)`

Reply:
Done.